src/app/pipes/timedisplay.ts

```diff
@@ -0,0 +1,41 @@
+import { PipeTransform, Pipe } from '@angular/core';
+
+
+@Pipe({name: 'timeDisplay'})
+export class TimeDisplayPipe implements PipeTransform {
+    transform(timeInSeconds: number, shortened = false): string {
+        if (timeInSeconds / 86400 >= 1) {
```

Comment:
There's a lot of duplication of code here. How about something like this (untested)?

```javascript
const timeInSeconds = 86400 * 5 + 3600 * 3 + 3 * 60 + 7
const shortened = true;

let timeInSecondsLeft = timeInSeconds;
const days = Math.floor(timeInSecondsLeft / 86400);
timeInSecondsLeft = timeInSecondsLeft % 86400;
const hours = Math.floor(timeInSecondsLeft / 3600);
timeInSecondsLeft = timeInSecondsLeft % 3600;
const minutes = Math.floor(timeInSecondsLeft / 60);
const seconds = timeInSecondsLeft % 60;

let time = '';
// TODO: Convert all these if statements to use a helper function
if (days > 0) {
    time += days;
    if (shortened === true) {
        time += ' d';
    } else {
        time += ' days';
    }
}

if (hours > 0) {
    if (time !== '') {
        time += ', ';
    }
    time += hours;
    if (shortened === true) {
        time += ' h';
    } else {
        time += ' hours';
    }
}

if (minutes > 0) {
    if (time !== '') {
        time += ', ';
    }
    time += minutes;
    if (shortened === true) {
        time += ' m';
    } else {
        time += ' minutes';
    }
}

if (days === 0 && seconds > 0) {
    if (time !== '') {
        time += ', ';
    }
    time += seconds;
    if (shortened === true) {
	      time += ' s';
    } else {
	      time += ' seconds';
    }
}

console.log(time);
```